src/net/sourceforge/kolmafia/request/PlaceRequest.java

```diff
@@ -138,262 +128,211 @@ public static void parseResponse(final String urlString, final String responseTe
       action = "";
     }
 
-    if (place.equals("arcade")) {
-      ArcadeRequest.parseResponse(urlString, responseText);
-    } else if (place.startsWith("batman")) {
-      BatFellowRequest.parseResponse(urlString, responseText);
-    } else if (place.equals("campaway")) {
-      CampAwayRequest.parseResponse(urlString, responseText);
-    } else if (place.equals("chateau")) {
-      ChateauRequest.parseResponse(urlString, responseText);
-    } else if (place.equals("crimbo16m")) {
-      // A Meditation Mat
-    } else if (place.equals("crimbo22")) {
-      if (action.equals("crimbo22_engine") || action.equals("c22_locobox")) {
-        // This redirects to a fight until you have defeated the boss
-        // If we are here now, we have done that.
-        //
-        // You've already defeated the Trainbot boss. There's nothing else of interest in that
-        // locomotive. Not even an explanation for why the train is still running!
-        Preferences.setBoolean("superconductorDefeated", true);
-      }
-    } else if (place.equals("desertbeach")) {
-      if (action.equals("db_nukehouse")) {
-        if (responseText.contains("anticheese")) {
-          Preferences.setInteger("lastAnticheeseDay", KoLCharacter.getCurrentDays());
-        }
-      }
-    } else if (place.equals("drip")) {
-      // You can't enter The Drip without wearing one of those harnesses Jeremy told you about.
-      if (responseText.contains("otherimages/drip/hall.gif")) {
-        Preferences.setBoolean("drippingHallUnlocked", true);
-      }
-    } else if (place.equals("dripfacility")) {
-      if (action.equals("drip_jeremy")) {
-        // You show Jeremy the big snail shell you found.
-        if (responseText.contains("You show Jeremy the big snail shell you found")) {
-          Preferences.setBoolean("drippyShieldUnlocked", true);
-          ResultProcessor.removeItem(ItemPool.DRIPPY_SNAIL_SHELL);
-        }
-        // Oooh, that big fingernail, that's interesting.
-        // I'll send it up to the Armory right away.
-        // In the meantime, our scouts have cleared the way to another section of the Drip.
-        // There's some kind of big building there. Go check it out, would you?
-        if (responseText.contains("Oooh, that big fingernail, that's interesting")) {
+    switch (place) {
```

Comment:
next time you do a large-scale refactoring like this, can you do it in a separate commit (or even PR)? It's hard to see the relevant changes.

Also can this be a switch expression? I don't see any fall-through.

Reply:
Yeah I wanted to do that and forgot that I had it sitting there when I came back to work more, so it was sort of locked in without me repeating work.

Reply:
What do you mean by your second question?

Reply:
it's currently `switch` with `case:`. It could be `switch` with `case ->`, and that would save me having to track every case to ensure there's a `break`.

Reply:
Done! Only didn't do it that way because IntelliJ didn't offer it to me automatically haha